pkg/controller/common/util.go

```diff
@@ -612,6 +618,12 @@ func GetPodsUsingPVCs(ctx context.Context, c client.Client, namespace string, na
 						// all mounts must be ro
 						addPod = false
 					}
+					if strings.HasSuffix(pod.Name, common.ClonerSourcePodNameSuffix) && pod.Labels != nil &&
+						pod.Labels[common.CDIComponentLabel] == common.ClonerSourcePodName {
+						// Host assisted clone source pod only reads from source
+						// But some drivers disallow mounting a block PVC ReadOnly
+						addPod = false
```

Comment:
Several consecutive conditions lead to the same result (false addPod), would it be possible to clean-up this a little so we have it more condenssed? Maybe something like a `shouldAddPod` function? Not a must but I think this code is getting hard to follow.

Reply:
Yeah the whole function can be refactored but I prefer to stay on the safe side since this is going to backport all the way to 0.55